cli/cmd/profile.go

```diff
@@ -94,12 +96,24 @@ func newCmdProfile() *cobra.Command {
 		Args: cobra.ExactArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
 			options.name = args[0]
+			clusterDomain := defaultClusterDomain
 
 			err := options.validate()
 			if err != nil {
 				return err
 			}
 
+			// --ignore-cluster triggers offline profile generation and clusterDomain here is cluster.local
+			if options.ignoreCluster {
```

Comment:
Can we flip this around so that it's something like:

```
if !options.ignoreCluster {
  // BUILD API CLIENT HERE
  // FETCH AND SET CLUSTER DOMAIN HERE
}

if options.template {
... 
```

This we way don't have to repeat the main render calls.

Reply:
Okay! I am working on this.